src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/SetAzureApiManagementBackend.cs

```diff
@@ -0,0 +1,128 @@
+        [ValidateNotNullOrEmpty]
+        [ValidateSet("http", "soap")]
+        public String Protocol { get; set; }
+
+        [Parameter(
+            ValueFromPipelineByPropertyName = true,
+            Mandatory = false,
+            HelpMessage = "Runtime Url for the Backend. This parameter is optional.")]
+        [ValidateNotNullOrEmpty]
+        [ValidateLength(1, 2000)]
+        public String Url { get; set; }
+
+        [Parameter(
+            ValueFromPipelineByPropertyName = true,
+            Mandatory = false,
+            HelpMessage = "Original uri of entity in external system backend points to. This parameter is optional.")]
+        public String ResourceId { get; set; }
+
+        [Parameter(
+            ValueFromPipelineByPropertyName = true,
+            Mandatory = false,
+            HelpMessage = "Backend Title. This parameter is optional.")]
+        [ValidateLength(1, 300)]
+        public String Title { get; set; }
+
+        [Parameter(
+            ValueFromPipelineByPropertyName = true,
+            Mandatory = false,
+            HelpMessage = "Backend Description. This parameter is optional.")]
+        [ValidateLength(1, 2000)]
+        public String Description { get; set; }
+
+        [Parameter(
+            ValueFromPipelineByPropertyName = true,
+            Mandatory = false,
+            HelpMessage = "Whether to Skip Certificate Chain Validation when talking to the Backend. This parameter is optional.")]
+        public bool? SkipCertificateChainValidation { get; set; }
+
+        [Parameter(
+            ValueFromPipelineByPropertyName = true,
+            Mandatory = false,
+            HelpMessage = "Whether to skip Certificate Name Validation when talking to the Backend. This parameter is optional.")]
+        public bool? SkipCertificateNameValidation { get; set; }
+
+        [Parameter(
+            ValueFromPipelineByPropertyName = true,
+            Mandatory = false,
+            HelpMessage = "Credential to be used when talking to the Backend. This parameter is optional.")]
+        public PsApiManagementBackendCredential Credential { get; set; }
+
+        [Parameter(
+            ValueFromPipelineByPropertyName = true,
+            Mandatory = false,
+            HelpMessage = "Proxy Server details to be used while sending request to the Backend. This parameter is optional.")]
+        public PsApiManagementBackendProxy Proxy { get; set; }
+
+        [Parameter(
+            ValueFromPipelineByPropertyName = true,
+            Mandatory = false,
+            HelpMessage = "If specified then instance of " +
+                          "Microsoft.Azure.Commands.ApiManagement.ServiceManagement.Models.PsApiManagementBackend type " +
+                          " representing the modified backend will be written to output.")]
+        public SwitchParameter PassThru { get; set; }
+
+        public override void ExecuteApiManagementCmdlet()
+        {
+            Client.BackendSet(
```

Comment:
#Fixed

---
In reply to: [104577504](https://github.com/Azure/azure-powershell/pull/3586#discussion_r104577504) [](ancestors = 104577504)